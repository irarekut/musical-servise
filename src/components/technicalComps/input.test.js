/* eslint-disable no-undef */

import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Input from './input'

describe('<Input />', () => {
  describe('Snapshots tests', () => {
    it('should render successfully and match snapshot', () => {
      const { container } = render(<Input>e-mail</Input>)

      expect(container).toMatchSnapshot()
    })
  })
  describe('Attributes test', () => {
    it('should set type attribute', () => {
      const type = 'email'
      render(<Input type={type} />)

      expect(screen.getByRole('textbox')).toHaveAttribute('type', type)
    })
  })
  describe('OnChange test', () => {
    it('should call onChange prop', () => {
      const cb = jest.fn()

      render(<Input onClick={cb} />)
      const input = screen.getByRole('textbox')

      fireEvent.change(input, { target: { value: 'testVal' } })
      expect(input.value).toBe('testVal')
    })
  })
})
