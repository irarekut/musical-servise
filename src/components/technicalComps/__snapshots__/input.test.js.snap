// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots tests should render successfully and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe bRLOeJ"
  >
    <input
      class="sc-gswNZR fbvIkD"
      value=""
    />
    <span
      class="sc-dkrFOg PdUkF"
    />
  </div>
</div>
`;
